react/ActionMenu/index.jsx

```diff
@@ -1,160 +1,20 @@
-    const minimumCloseVelocity = 0.6
-
-    let currentGestureProgress = null
-
-    this.gesturesHandler.on('panstart', () => {
-      this.turnTransitionsOff()
-      currentGestureProgress = 0
-    })
-
-    this.gesturesHandler.on('pan', e => {
-      currentGestureProgress = e.deltaY / maximumGestureDistance
-      this.applyTransformation(currentGestureProgress)
-    })
-
-    this.gesturesHandler.on('panend', e => {
-      // Dismiss the menu if the swipe pan was bigger than the treshold,
-      // or if it was a fast, downward gesture
-      let shouldDismiss =
-        e.deltaY / maximumGestureDistance >= minimumCloseDistance ||
-        (e.deltaY > 0 && e.velocity >= minimumCloseVelocity)
-
-      if (shouldDismiss) {
-        if (currentGestureProgress >= 1) {
-          // Menu is already hidden, close it right away
-          this.close()
-        } else {
-          this.turnTransitionsOn()
-          this.animateClose()
-        }
-      } else {
-        this.turnTransitionsOn()
-        this.applyTransformation(0)
-      }
-    })
-  }
-
-  /**
-   * Applies a css trasnform to the element, based on the progress of the gesture
-   * @param  {Float} progress - Amount of progress between 0 and 1
-   */
-  applyTransformation(progress) {
-    // constrain between 0 and 1.1 (go a bit further than 1 to be hidden completely)
-    progress = Math.min(1.1, Math.max(0, progress))
-    this.menuNode.style.transform = 'translateY(' + progress * 100 + '%)'
-  }
-
-  animateClose = () => {
-    this.setState({ closing: true })
-
-    // we need to transition the menu to the bottom before dismissing it
-    const close = once(() => {
-      this.menuNode && this.menuNode.removeEventListener('transitionend', close)
-      this.close()
-    })
-
-    this.menuNode &&
-      this.menuNode.addEventListener('transitionend', close, false)
-    // in case transitionend is not called, for example if the element is removed
-    setTimeout(close, TRANSITION_DURATION)
-
-    this.applyTransformation(1.1)
-  }
-
-  close = () => {
-    this.setState({ closing: true })
-    this.props.onClose()
-  }
-
-  handleMenuRef = menu => {
-    this.menuNode = menu
-  }
-
-  handleWrapperRef = wrapper => {
-    this.wrapperNode = wrapper
-  }
-
-  render() {
-    const { children, className } = this.props
-    const { closing } = this.state
-    return (
-      <div className={className} ref={this.handleWrapperRef}>
-        <Overlay
-          style={{ opacity: closing ? 0 : 1 }}
-          onClick={this.animateClose}
-          onEscape={this.animateClose}
-        >
-          <div className={styles['c-actionmenu']} ref={this.handleMenuRef}>
-            {children}
-          </div>
-        </Overlay>
-      </div>
-    )
-  }
-}
+const ActionMenu = ({ children, className, onClose }) => (
+  <div className={className}>
+    <BottomDrawer onClose={onClose}>
+      <div className={styles['c-actionmenu']}>{children}</div>
+    </BottomDrawer>
+  </div>
+)
```

Comment:
It seems we have no control on the inner actionMenu className.

Why is the wrapping div necessary ? Could the className be put on the BottomDrawer for the structure to be simpler ?

Reply:
You may have seen that in the more recent code, the inner menu can be wrapped either in `BottomDrawer` or `ClockAwayListener`. We could pass the props to `BottomDrawer`, but `ClickAwayListener` absolutely needs a wrapper. I think injecting the classname in 2 different ways would make the structure more complicated.